docs/Developers/Design/CRD_design.md

```diff
@@ -0,0 +1,438 @@
+This document describes the set of [Custom Resource Definitions
+(CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+that will be used to configure a GCS Gluster cluster. The actual implementation
+of the resources described here will be phased in during development. The
+purpose of this document is to provide the overall structure, ensuring the end
+result provides necessary configurability in a user-friendly manner.
+
+# Overview
+
+A single Gluster operator may control one or more individual Gluster clusters.
+Each cluster could be either hosted within the Kubernetes cluster as a set of
+pods (a converged configuration) or the Gluster nodes could be running on hosts
+outside the Kubernetes cluster (referred to as independent mode). The
+capabilities of the operator differ significantly between these two modes of
+deployment, but the same set of CRDs should be used for both where possible.
+
+A given Gluster cluster is defined by several different Custom Resources (CRs)
+that form a hierarchy. At the top level is the "cluster" CR that describes
+cluster-wide configuration options such as the "name" for the cluster, the TLS
```

Comment:
@JohnStrunk  can you provide some details about "name" ?

Reply:
The "name" is the `name` field in the Cluster (soon to be GlusterCluster) CR. It would show up as the Service name to which CSI connects to in order to access this Gluster cluster.